File: src/main/java/com/hsmchurch/app/video/ui/request/YoutubeVideoInfo.java
package com.hsmchurch.app.video.ui.request;

import com.hsmchurch.app.video.support.DescriptionParseResult;
import com.hsmchurch.app.video.support.DescriptionParser;
import lombok.Builder;
import lombok.Value;

@Builder
@Value
public class YoutubeVideoInfo {

    private final String id;
    private final String publishedAt;
    private final String title;
    private final String description;
    private final ThumbNailForm thumbNail;

    public DescriptionParseResult parseDescription(final DescriptionParser parser){
        return parser.parse(description);
    }

}

File: src/main/java/com/hsmchurch/app/video/ui/request/YoutubeVideoInfo.kt
package com.hsmchurch.app.video.ui.request

import com.hsmchurch.app.video.support.DescriptionParseResult
import com.hsmchurch.app.video.support.DescriptionParser
import lombok.Builder
import lombok.Value

data class YoutubeVideoInfo(val id: String,
                            val publishedAt: String,
                            val title: String,
                            val description: String?,
                            val thumbNail: ThumbNailForm) {

    fun parseDescription(parser: DescriptionParser): DescriptionParseResult {
        return parser.parse(description!!)
    }

}
